src/Action/SetObjectFieldValueAction.php

```diff
@@ -113,34 +119,25 @@ public function __invoke(Request $request): JsonResponse
             $propertyPath = new PropertyPath($field);
         }
 
-        // Handle date type has setter expect a DateTime object
-        if ('' !== $value && 'date' === $fieldDescription->getType()) {
-            $value = new \DateTime($value);
-        }
+        if ('' === $value) {
+            $this->pool->getPropertyAccessor()->setValue($object, $propertyPath, null);
+        } else {
+            $dataTransformer = $this->getFieldDataTransformer($fieldDescription, $admin->getModelManager());
 
-        // Handle boolean type transforming the value into a boolean
-        if ('' !== $value && 'boolean' === $fieldDescription->getType()) {
-            $value = filter_var($value, FILTER_VALIDATE_BOOLEAN);
-        }
+            if ($dataTransformer instanceof DataTransformerInterface) {
+                $value = $dataTransformer->reverseTransform($value);
+            }
 
-        // Handle entity choice association type, transforming the value into entity
-        if ('' !== $value
-            && 'choice' === $fieldDescription->getType()
-            && null !== $fieldDescription->getOption('class')
-            && $fieldDescription->getOption('class') === $fieldDescription->getTargetEntity()
-        ) {
-            $value = $admin->getModelManager()->find($fieldDescription->getOption('class'), $value);
-
-            if (!$value) {
+            if (!$value && 'choice' === $fieldDescription->getType()) {
```

Comment:
@core23 i am very confused by the tight binding to the field type here. I understand why this is done, but it still looks like a dirty hack.